### PR TITLE
add support for Lambda file systems

### LINKED ISSUE #13780: AWS Lambda support for Amazon Elastic File System
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS Lambda customers can now enable functions to access Amazon Elastic File System (Amazon EFS). 

### New or Affected Resource(s)

* [`aws_lambda_function`](https://www.terraform.io/docs/providers/aws/r/lambda_function.html)

### Potential Terraform Configuration

```hcl
resource "aws_lambda_function" "example" {
  file_system_config {
    arn = "${aws_efs_access_point.example.arn}"

    local_mount_path = "/mnt/example"
  }
}
```

### References

[Announcement](https://aws.amazon.com/about-aws/whats-new/2020/06/aws-lambda-support-for-amazon-elastic-file-system-now-generally-/).
[Blog post](https://aws.amazon.com/blogs/aws/new-a-shared-file-system-for-your-lambda-functions/).
[Developer guide](https://docs.aws.amazon.com/lambda/latest/dg/services-efs.html).

Requires [AWS SDK v1.32.3](https://github.com/aws/aws-sdk-go/releases/tag/v1.32.3):
* https://github.com/terraform-providers/terraform-provider-aws/pull/13738

### PR DESCRIPTION
**Pull request by bnusunny**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note

ENHANCEMENTS:

* resource/aws_lambda_function: Add `file_system_config` attribute
* data-source/aws_lambda_function: Add `file_system_config` attribute

```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSLambdaFunction_FileSystemConfig'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSLambdaFunction_FileSystemConfig -timeout 120m
=== RUN   TestAccAWSLambdaFunction_FileSystemConfig
=== PAUSE TestAccAWSLambdaFunction_FileSystemConfig
=== CONT  TestAccAWSLambdaFunction_FileSystemConfig
--- PASS: TestAccAWSLambdaFunction_FileSystemConfig (2145.05s)
PASS
ok      github.com/terraform-providers/terraform-provider-aws/aws       2146.142s
...
```

Fixes #13780

### CODE REVIEW

**Comment by ewbankkit** [comment]
Closes https://github.com/terraform-providers/terraform-provider-aws/issues/13780.

File: aws/data_source_aws_lambda_function.go [hunk 1]
```diff
@@ -40,6 +40,22 @@ func dataSourceAwsLambdaFunction() *schema.Resource {
 					},
 				},
 			},
+			"file_system_config": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
```
Comment (ewbankkit) [inline aws/data_source_aws_lambda_function.go:46]: Please add `Computed: true,` as this attribute is "output-only".

File: aws/data_source_aws_lambda_function.go [hunk 2]
```diff
@@ -40,6 +40,22 @@ func dataSourceAwsLambdaFunction() *schema.Resource {
...
+					Schema: map[string]*schema.Schema{
+						"arn": {
+							Type:     schema.TypeString,
+							Required: true,
```
Comment (ewbankkit) [inline aws/data_source_aws_lambda_function.go:50]: Please add `Computed: true,` as this attribute is "output-only".
Remove `Required`.

File: aws/data_source_aws_lambda_function.go [hunk 3]
```diff
@@ -40,6 +40,22 @@ func dataSourceAwsLambdaFunction() *schema.Resource {
...
+						},
+						"local_mount_path": {
+							Type:     schema.TypeString,
+							Required: true,
```
Comment (ewbankkit) [inline aws/data_source_aws_lambda_function.go:54]: Please add `Computed: true,` as this attribute is "output-only".
Remove `Required`.

**Comment by ewbankkit** [comment]
@bnusunny Thanks for the contribution.
I've left a few minor initial review comments.

File: aws/data_source_aws_lambda_function.go (hunk 1 above)
Comment (bnusunny) [reply aws/data_source_aws_lambda_function.go:46]: Updated.

File: aws/data_source_aws_lambda_function.go (hunk 2 above)
Comment (bnusunny) [reply aws/data_source_aws_lambda_function.go]: Updated.

File: aws/data_source_aws_lambda_function.go (hunk 3 above)
Comment (bnusunny) [reply aws/data_source_aws_lambda_function.go]: Updated.

**Comment by bnusunny** [comment]
@gdavison Can you share which region the testing runs? I want to see what error you came across.

**Comment by gdavison** [comment]
We're running our tests in `us-west-2`, so it's one of
* the LA local zone
* an Outpost

The error was `Error: Error creating Lambda function: InvalidParameterValueException: The provided subnets contain availability zone Lambda doesn't support. Please use a different set of subnets.`

The workaround that we use is to filter the AZs:
```terraform
data "aws_availability_zones" "available" {
  state = "available"

  filter {
    name   = "opt-in-status"
    values = ["opt-in-not-required"]
  }
}
```

**Comment by bnusunny** [comment]
Oh.  You have access to both local zone and an outpost. Very cool!

**Pull request merged**